src/ui/content/register.tsx

```diff
@@ -1,48 +1,101 @@
-import React from 'react'
+import React, { useState } from 'react'
+import gql from 'graphql-tag'
+import { useMutation } from '@apollo/react-hooks'
 
-type MyProps = { }
-type MyState = { email: string, name: string, password: string, password2: string}
+interface UserCreateInput {
+  name: string,
+  email: string,
+  password: string,
+  age: number,
+}
 
-class RegForm extends React.Component<MyProps, MyState> {
-  constructor(props: any) {
-    super(props);
-    this.state = {email: '', name:'', password:'', password2:''};
+const REG_M = gql`
+  mutation Register($data: UserCreateInput!) {
+    signupUser(data: $data) {
+      name
+    }
+  }
+`
+function RegForm() {
+  const [email, setEmail] = useState('')
+  const [name, setName] = useState('')
+  const [password, setPassword] = useState('')
+  const [password2, setPassword2] = useState('')
 
-    this.handleChange = this.handleChange.bind(this);
-    this.handleSubmit = this.handleSubmit.bind(this);
+  //why cannot I use data:response here? is it a new "data"?
```

Comment:
when I defined the mutation, I used the name "response" for data. Is this a different data from that?